core/validation/pre_transformation_validate.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+
+from core.const import PreTransformationCheck
+from core.exceptions import ValidationError
+from core.validation.failures.user import (
+    UnauthorisedSubmissionFailure,
+    WrongInputFailure,
+)
+from core.validation.pre_transformation_validation_schema import TF_ROUND_3, TF_ROUND_4
+
+
+def pre_transformation_validations(
+    workbook: dict[str, pd.DataFrame],
+    reporting_round: int,
+    auth: dict,
+):
+    """Performs pre-transformation validations based on the provided schema.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the submission.
+    :param reporting_round: The reporting round for which the validations are performed.
+    :param auth: A dictionary containing authorization information.
+    :raises ValidationError: If any of the validation functions catch a validation error.
+    """
+    if reporting_round in [1, 2]:
+        return  # do not do pre-transformation validation on historical data
+
+    round_to_schema = {3: TF_ROUND_3, 4: TF_ROUND_4}
+
+    schema = round_to_schema[reporting_round]
```

Comment:
Can set as a dictionary in ```validation/const.py``` (I originally brought it into the function as was causing a circular import error on ```core/const.py```.

Presumably reporting round will reset to 1 for every new fund, and so we'll have to have a two keys on the dictionary look-up in the future, but without entirely knowing what that looks like it's hard to be more specific now.

Reply:
Yep I agree we'll end up adding `fund` as a parameter later but if we abstract this logic out then it will mean less changes to this function when we do that.. perhapse this is fine for now though - we'll have to make a few more changes here when those changes come anyway